src/packages/auth/functions.ts

```diff
@@ -212,37 +220,37 @@ export function getDefaultRoleForAllowedRoles(allowedRoles: string[]): string {
     }
   }
 
-  throw new Error(`Fatal error, not able to find a matching default role within the following auth roles: ${allowedRoles}`);
+  throw new Error(
+    `Fatal error, not able to find a matching default role within the following auth roles: ${allowedRoles}`,
+  );
 }
 
 export function mapGroupsToRoles(groupList: string[]): UserRoles {
   const { DEFAULT_ROLE, ALLOWED_ROLES, AUTH_GROUP_ROLE_MAPPINGS } = getEnv();
 
-  let default_role = DEFAULT_ROLE[0];
-  let allowed_roles  = ALLOWED_ROLES;
-
   // use auth group -> aerie role mappings if set
   if (JSON.stringify(AUTH_GROUP_ROLE_MAPPINGS) !== '{}') {
     const mappedGroupMembership = getGroupsWithMappings(groupList);
     if (mappedGroupMembership.length > 0) {
-      allowed_roles = getAllAllowedRolesForAuthGroups(mappedGroupMembership);
-      default_role = getDefaultRoleForAllowedRoles(allowed_roles);
+      const allowed_roles = getAllAllowedRolesForAuthGroups(mappedGroupMembership);
+      return {
+        allowed_roles,
+        default_role: getDefaultRoleForAllowedRoles(allowed_roles),
+      }
```

Comment:
Return is missing for the `else` case of this if

Reply:
If `mappedGroupMembership` is empty, allowed roles and default roles will also be empty. I think we can handle this error at a broader level (i.e. `loginSSO`), and return a failed login if `allowed_roles` is empty after checking the DB.

Reply:
Although that does create an issue of "stale" membership, e.g. LDAP membership might change external to Aerie, but our DB is the source of truth once a user is created.

Reply:
Maybe we can just take the intersection of DB `allowed_roles` and auth group mapped `allowed_roles`?

Reply:
If the user signing in isn't new, the DB check will pass, because it'll be using the DB data, not the empty data. Additionally, the issue here is that the method isn't explicitly returning empty, it's returning nothing in the else case

Reply:
RE: Taking the intersection/updating the DB to reflect changes in LDAP, while I'm not against it, I'd say check we should check in with @dandelany and/or @joswig, as that's an additional behavior change.

Reply:
Ah wrote those comments based on local changes I had where I removed the `length > 0` check entirely 🤦

That way, we'll forward an empty `allowed_roles` list if `mappedGroupMembership` is empty, and the downstream auth adapter can handle that case however they want (e.g. continue to check user in DB, return failed login, etc)